CenterForOpenScience/osf.io
# [PLAT-961] Add change password functionality to APIv2

## Purpose

As part of emberification/emberization we need to give users the ability to change their passwords, like they can on the account settings page. This allows them to do that.

## Changes

- Add view and serializer for password changing
- add tests for above view

## QA Notes

Something like the following should be a good test:
```python

data = {
    "data": {
        "type": "user-password",
        "id": "5pq94",
        "attributes": {
            "existing_password": "old-password",
            "new_password": "new-password"
        }
    }
}

resp = requests.post('http://localhost:8000/v2/users/5pq94/settings/password/', data=json.dumps(data), headers={'Authorization': 'Bearer  {token}', 'Content-type': 'application/json'})

print(resp.status_code)
```

## Documentation

No user-facing changes

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-961
